zstackctl/zstackctl/ctl.py

```
@@ -9127,6 +9164,14 @@ def run(self, args):
         ssh_key = ctl.zstack_home + "/WEB-INF/classes/ansible/rsaKeys/id_rsa.pub"
         private_key = ssh_key.split('.')[0]
         inventory_file = ctl.zstack_home + "/../../../ansible/hosts"
+
+        # pre-check local node before stopping any node
+        if not args.skip_precheck:
+            linux.rm_dir_force("/tmp/zstack_upgrade.lock")
```

Ruff (0.15.7)

[error] Probable insecure usage of temporary file or directory: "/tmp/zstack_upgrade.lock"

(S108)